src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Validation/ValidationVisitor.cs

```diff
@@ -192,6 +196,27 @@ protected virtual bool Visit(ModelMetadata metadata, string key, object model)
                 return true;
             }
 
+            if (++ValidationDepth > MaxValidationDepth)
+            {
+                // Non cyclic but too deep an object graph.
+                string displayString;
+                switch (metadata.MetadataKind)
```

Comment:
Will need to handle `Metadata.Kind.Parameter` on the off chance someone sets the limit to `0`.